Add null checks on  role mapping

Amoncy opened the pull request:
Hi @andreaTP and @EricWittmann,

This PR closes #3776 .

I have added the missing null checks on role mapping. Could you please review the changes?
If the changes are correct, I'll proceed with implementing the test cases.

Thank you

EricWittmann commented on app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java:
```diff
@@ -337,6 +337,9 @@ public RoleMapping getRoleMapping(String principalId) {
     @RoleBasedAccessApiOperation
     public void updateRoleMapping(String principalId, UpdateRole data) {
+        requireParameter("principalId", principalId);
+        requireParameter("roleName", data.getRole().name());
```
But what if the `role` field is not present or is empty?  We need to check for that.

Also need a unit test I think.

andreaTP replied:
Correct, here the NPE is thrown by `data.getRole()` returning `null`.
And I think we need tests 👍 , I hope there is already some infrastructure to mock rbac etc.

Amoncy replied:
Thank you for the review @EricWittmann and @andreaTP .
 I'll add null check for Role and add Unit Test as well.

Amoncy commented:
Hi,

I have added the check for Role and added the corresponding Unit Tests.

Please Note: I have removed the check for "name" as I realised while going through the code, that it belongs to enum which doesn't have a possibility of a null or empty value. Hence, I haven't added any test case for the same.

Please let me know your views on this.
Thank you

andreaTP commented on app/src/test/java/io/apicurio/registry/rbac/AdminResourceTest.java:
```diff
@@ -720,6 +720,16 @@ public void testRoleMappings() throws Exception {
                 .body("error_code", equalTo(404))
                 .body("message", equalTo("No mapping for principal 'UnknownPrincipal' and role 'READ_ONLY' was found."));
 
+        //Update a mapping with null RoleType
```
👍

EricWittmann commented:
Thanks @Amoncy - this looks good.  @andreaTP is right that we need to review basically our entire REST layer and add input checks.  I think that might be tracked already in GH as a separate issue.  If not we should add it to the 3.0 roadmap.  But it's out of scope for this PR for sure.  :)

Amoncy commented on app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java:
```diff
@@ -337,6 +337,8 @@ public RoleMapping getRoleMapping(String principalId) {
     @RoleBasedAccessApiOperation
     public void updateRoleMapping(String principalId, UpdateRole data) {
+        requireParameter("principalId", principalId);
+        requireParameter("role", data.getRole());
```
Thank you @andreaTP for the review. 
I'll look for related beginner friendly issues and add the checks there as well.

Amoncy commented:
> Thanks @Amoncy - this looks good. @andreaTP is right that we need to review basically our entire REST layer and add input checks. I think that might be tracked already in GH as a separate issue. If not we should add it to the 3.0 roadmap. But it's out of scope for this PR for sure. :)

Thanks Eric for the review.
